packages/pyright-internal/src/analyzer/binder.ts

```diff
@@ -4331,6 +4333,10 @@ export class Binder extends ParseTreeWalker {
     private _addSyntaxError(message: string, textRange: TextRange) {
         return this._fileInfo.diagnosticSink.addDiagnosticWithTextRange('error', message, textRange);
     }
+
+    private _ignoreUntypedModule(module: string) {
+        return this._fileInfo.diagnosticRuleSet.allowedUntypedLibraries.some(x => module.startsWith(x));
```

Comment:
this should check each mofule part (separated by `.`) to prevent a module like `foobar` from being matched by `allowedUntypedLibraries = ["foo"]`

Reply:
oops !